packages/app/src/cli/services/dev.ts

```diff
@@ -135,6 +140,15 @@ async function dev(options: DevOptions) {
   const extensionsIds = prodEnvIdentifiers.app === apiKey ? envExtensionsIds : {}
   localApp.extensions.ui.forEach((ext) => (ext.devUUID = extensionsIds[ext.localIdentifier] ?? ext.devUUID))
 
+  const {extensionIds: remoteExtensions} = await ensureDeploymentIdsPresence({
```

Comment:
Could we move this piece of code inside the following 

```ts
if (nonPreviewableExtensions.length > 0) 
```

it seems that `remoteExtensions` is only used inside thus we could avoid making API requests